### DISCUSSION TITLE
Scale Servings of a recipe

### DISCUSSION THREAD

**Discussion by johamins** [mealie-recipes/mealie#1161] [Feature Request]
**Before submitting this feature request I have**
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
I'd like a function to scale up the servings of a recipe. For example if a recipe is intended for 4 servings, but I have 10 persons for a meal.

**Suggest A Solution**
Maybe just a input field for the amount of servings, so that you can enter your number and it will calculate the ingredients before adding to a meal plan or shopping list.

**Comment by hay-kot** [comment]
FYI, this has been implemented in the beta.

https://beta.mealie.io

**Reply by johamins** [reply]
I think that's not really what I mean ( or I just didn't found it.)
What I saw was for example in this recipe: https://beta.mealie.io/recipe/peanut-butter-cookies-iv
I can scale up the amount of cookies. But it will tell me for example to use 3 times 1/2 cup of butter instead of 1 1/2 cups of butter.
And on other recipes I don't see anything for scaling, like for example here: https://beta.mealie.io/recipe/chickpea-salad. It just has 4 servings, but I'd like it to calculate it for 6 servings for example.

**Comment by coolkau** [comment]
I get, that this is quite difficult to implement to such a generalistic ingredients field, but this really would be a killer feature.

**Comment by Duchessy** [comment]
I would also really like to see this feature.
At the moment you can enter a serving size but with the "+" and "-" buttons you can only multiply the original serving size so from 4 to 8 to 12 and so on.
In most other recipe apps the "+" and "-" buttons only ad or subtract one serving so from 4 to 5 or you can enter a specific number of servings and the amount of ingredients is calculated.

A possible solution for me could also be to first adjust the scale manually and for the "+" and "-" to multiply the adjusted scale.
For example: I have a recipe with 4 servings. Then I adjust the scale to 0,25 so I only get the ingeredients for 1 serving. Now I use the "+" button and the serving size is doubled to 2 (At the moment it would now show 5 (1+4) servings.

**Reply by fleshgolem** [reply]
While I agree that the way this element works really not that great, the proposed solution is IMHO pretty hard to comprehend and still kind of clunky.
Here is maybe an idea that can keep the best of both words:

Add two new buttons. The + and - buttons now add/subtract a serving while the new buttons add/substract a batch (current behaviour)
![Screenshot_20230226_211647](https://user-images.githubusercontent.com/657606/221434886-29050e7c-d08d-4476-a7b9-56021bf24792.png)

The popup could maybe changed to something like this, with both inputs auto-updating each other (this is just a really quick pass, i suck at design)
![Screenshot_20230226_211924](https://user-images.githubusercontent.com/657606/221435095-b283efbb-ea5c-4306-9665-14507e3bf1ea.png)
Personally I would just get rid of the scale input altogether, but I know why HK finds it useful, so that is probably a non-starter :)

**Reply by Duchessy** [reply]
Yes, I really like your solution!
This would totaly work for me!

**Comment by nick-dawson** [comment]
I'm commenting to show support for this feature in general. We were using the app Whisk to track recipes and it has a nice feature to adjust ingredients based on changing the serving size by increments of one (screenshot below). Changes in serving size are then reflected by respective changes to ingredient quantities. It's really the only feature we miss after moving to Mealie. We typically make meals of 10 servings each week which usually isn't a multiple of the base serving so increments of +/- 1 would be super helpful for our family. If +/- 1 increments are too difficult to implement, we will do the math manually and keep using Mealie anyway because it's awesome. Thanks for all the work you do!

![IMG_5817](https://user-images.githubusercontent.com/32081605/224502322-f657141b-934f-4243-ab91-20028939b0a5.jpeg)

**Comment by michael-genson** [comment]
Similar discussion here: https://github.com/mealie-recipes/mealie/issues/1390

**Reply by GideonWolfe** [reply]
@michael-genson wow, it's funny how recently active this issue is, I literally installed `mealie` for the first time today coming from the Nextcloud Cookbook app, and this was the first feature I noticed missing pretty much immediately and I'm quite surprised since its such a useful feature and common usecase! Loving the application so far though.
